test/nested-properties.test.js

```diff
@@ -0,0 +1,70 @@
+// Copyright IBM Corp. 2013,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// This test written in mocha+should.js
+'use strict';
+
+/* global getSchema:false, connectorCapabilities:false */
+var should = require('./init.js');
+
+var db, Model;
+
+describe('basic-querying', function() {
+  before(function(done) {
+    var modelDef = {
+      'nested': {
+        'duration': Number,
+        'happenedOn': Date,
+      },
+    };
+
+    db = getSchema();
+    Model = db.define('Model', modelDef);
+    db.automigrate(done);
+  });
+
+  describe('ping', function() {
+    it('should be able to test connections', function(done) {
+      db.ping(function(err) {
+        should.not.exist(err);
+        done();
+      });
+    });
+  });
+
+  describe('find', function() {
+    before(function(done) {
+      db = getSchema();
+      Model.destroyAll(done);
+    });
+
+    it('should query by nested Date: found', function(done) {
+      Model.create({
+        nested: {
+          duration: 2,
+          happenedOn: '2017-10-30T06:07:31.805Z',
+        },
+      }, function(err, m) {
```

Comment:
We prefer to use Promises in the new tests we are writing, could you please rewrite this test?

```js
it('should query by nested Date: found', function() {
  return Model.create({/*...*/})
    .then(created => {
      should.exist(created.id);
      return Model.find({/*...*/});
    })
    .then(found => {
      should.exist(found);
      found.should.not.be.empty();
      found[0].should.be.an.instanceOf(Model);
    });
```